docs/product/roadmap.rst

```diff
@@ -1,6 +1,237 @@
 
-*Coming soon.*
+We have many more features planned for Mathesar, some of them are quite essential. See :doc:`Future Features <future>`.
+
+Installation and Configuration
+------------------------------
+Users should be able to:
+
+- Follow provided instructions to install Mathesar on a server.
+
+  - The installation process should only install PostgreSQL if needed.
+- Access existing PostgreSQL databases via Mathesar using existing PostgreSQL user credentials.
+
+  - Existing databases should reflect all columns and types correctly in the user interface.
+- Set up a PostgreSQL server automatically if none exists.
+- Create a new database from scratch.
+- Create an initial user if needed.
+- Configure sending email (for password resets, notifications)
+
+User Settings
+-------------
+Users should be able to:
+
+- Configure Mathesar to display databases in either technical or non-technical mode.
+
+  - "Technical mode" is going to use PostgreSQL terms for everything (e.g. schemas are schemas, not applications)
+
+CRUD For Tables & Schemas
+-------------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Create a new table
+
+  - in an existing schema
+  - in a new schema
+- Create a table via importing data via
+   - CSV
+   - TSV
+   - copy/paste from a spreadsheet
+- View a table and see what type each column is.
+- Edit a table name
+- Edit a schema name
+- Edit the data in their table using a spreadsheet-like interface
+
+  - e.g. fill down cells by dragging a corner
+- Delete a table
+- Delete a schema
+
+At this point, all columns are displayed and edited as strings. We will support reflecting types from the database, but we're not going to be doing anything with those types yet.
+
+Filtering, Sorting, Grouping
+----------------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Filter the table by rules applied to data in various fields.
+- Sort the table by various fields.
+- Group results by the first character of any given field.
+- Group results by the first word of any given field.
+- Apply multiple filters, groups, and/or sorts at once.
+
+Multiple Types Support
+----------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Change the type of data per-column to any type supported by Postgres or PostGIS
+
+  - The data in the column will be validated.
+- Import data via any supported import method and have the types of data automatically detected during the import process
+
+  - Types are suggestions and can be changed by the user.
+
+Boolean Type
+------------
+- Add new type (use Postgres type if possible):
+
+  - Boolean
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options by value (yes/no)
+
+New Text Types
+--------------
+- Add new types:
+
+  - Email
+  - URL
+- Autodetect these types during import
```

Comment:
I think there should be some agreed-upon minimum functionality to have a type be included in the UI.  Perhaps this should be part of that?